hera_mc/part_connect.py

```diff
@@ -292,6 +292,68 @@ def get_part_revisions(hpn, session=None):
     return revisions
 
 
+class Dubitable(MCDeclarativeBase):
+    """
+    A table to track the dubitable antennas.
+
+    start_gpstime:  start time of list
+    stop_gpstime:  stop time of list
+    ant_list:  list of "unapproved" antennas
+    """
+
+    __tablename__ = 'dubitable'
+
+    start_gpstime = Column(BigInteger, primary_key=True)
+    stop_gpstime = Column(BigInteger)
+    ant_list = Column(Text, nullable=False)
+
+    def __repr__(self):
+        return ('<{self.start_gpstime}>'.format(self=self))
+    #    return ('<{} dubitable entries>'.format(len(self.ant_list)))
+
+
+def update_dubitable(session=None, transition_gpstime=None, data=None):
+    """
+    update the database given a new list of dubitable antennas.
+    It will stop the previous list at transition_gpstime and start the new one
+    at transition_gpstime + 1sec
+
+    Parameters:
+    ------------
+    session:  db session to use
+    transition_gpstime:  gps time to make the change
+    data:  list of antennas
+    """
+
+    close_session_when_done = False
+    if session is None:
+        db = mc.connect_to_mc_db(None)
+        session = db.sessionmaker()
+        close_session_when_done = True
+
+    last_one = session.query(Dubitable).filter(Dubitable.stop_gpstime == None)
+    if last_one.count() == 1:  # Stop the previous valid list.
+        old_dubi = last_one.first()
+        old_dubi.stop_gpstime = transition_gpstime
+        session.add(old_dubi)
+    elif last_one.count() > 1:
+        raise ValueError('Too many open dubitable lists.')
+
+    new_dubi = Dubitable()
+    new_dubi.start_gpstime = transition_gpstime + 1
+    new_dubi.ant_list = ''
+    for a in data:
+        new_dubi.ant_list += (str(a) + ',')
+    new_dubi.ant_list = new_dubi.ant_list.strip(',')
+    session.add(new_dubi)
+    session.commit()
+    data_dict = {'start_gpstime': [transition_gpstime], 'ant_list:len': [len(data)]}
```

Comment:
Is the thinking that the list of antennas might be so long that we won't want to include it in the various logfiles and whatnot? That doesn't seem very likely to occur, to me.